hastory-data/src/Data/Hastory/Utils.hs

```diff
@@ -20,3 +31,17 @@ doCountsWith conv func = foldl go HM.empty
       where
         a Nothing = Just 1
         a (Just d) = Just $ d + func k
+
+dataBaseSpec :: SpecWith SqlBackend -> Spec
+dataBaseSpec = around withDatabase
+
+withDatabase :: ActionWith SqlBackend -> IO ()
+withDatabase func = runNoLoggingT $ withSqlConn openConn $ \conn -> do
```

Comment:
This makes _no_ sense.

https://hackage.haskell.org/package/persistent-sqlite-2.10.5/docs/Database-Persist-Sqlite.html#v:withSqliteConn
```
withSqliteConn :: (MonadUnliftIO m, MonadLogger m) => Text -> (SqlBackend -> m a) -> m a
```

That means the first argument, `openConn` is supposed to be a `Text`, but you pass in a `LogFunc -> IO SqlBackend`. What's going on here?

Reply:
You're not supposed to `open` the connection yourself.